lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -135,3 +137,21 @@ defaultRetryPolicy =
     limitRetriesByCumulativeDelay (60 * second) (constantDelay (1 * second))
   where
     second = 1000*1000
+
+-- | The 'nextBlocks' function should never return duplicate slots, blocks out
+-- of order, or blocks before the requested starting point. Doing so would break
+-- wallet invariants. This validation function should be used by 'NetworkLayer'
+-- implementations to ensure proper error handling.
```

Comment:
Actually, I believe that `nextBlocks` is also not allowed to return the block that corresponds to the starting point: it must come from a slot _**later**_ than the starting point.